crates/sui-types/src/base_types.rs

```diff
@@ -319,6 +319,15 @@ impl MoveObjectType {
             MoveObjectType_::Other(o) => s == o,
         }
     }
+
+    /// Returns the string representation of this object's type using the canonical display.
+    /// full address with 0x prefix
+    pub fn to_canonical_string(&self, with_prefix: bool) -> String {
+        let canonical_string = StructTag::from(self.clone())
+            .to_canonical_display(with_prefix)
+            .to_string();
+        canonical_string
+    }
 }
```

Comment:
Looks like the doc comment here is stale, and you don't need to bind `canonical_string` anymore.

Also, is it worth offering `to_canonical_display` here as well? (Only if there's a use for it).

Reply:
Yes let me update that

I don't know if it's possible to offer to_canonical_display - I would be returning a value reference a local variable (self.clone() converted into a StructTag) due to the signature of the custom display which is Display + '_.

Reply:
Yep good point on the lifetime issue!